plugin/pkg/admission/alwayspullimages/admission.go

```diff
@@ -85,23 +82,17 @@ func (*AlwaysPullImages) Validate(attributes admission.Attributes, o admission.O
 		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
 	}
 
-	for i := range pod.Spec.InitContainers {
-		if pod.Spec.InitContainers[i].ImagePullPolicy != api.PullAlways {
-			return admission.NewForbidden(attributes,
-				field.NotSupported(field.NewPath("spec", "initContainers").Index(i).Child("imagePullPolicy"),
-					pod.Spec.InitContainers[i].ImagePullPolicy, []string{string(api.PullAlways)},
-				),
-			)
-		}
-	}
-	for i := range pod.Spec.Containers {
-		if pod.Spec.Containers[i].ImagePullPolicy != api.PullAlways {
-			return admission.NewForbidden(attributes,
-				field.NotSupported(field.NewPath("spec", "containers").Index(i).Child("imagePullPolicy"),
-					pod.Spec.Containers[i].ImagePullPolicy, []string{string(api.PullAlways)},
-				),
-			)
+	var allErrs []error
+	pods.VisitContainersWithPath(&pod.Spec, func(c *api.Container, p *field.Path) {
+		if c.ImagePullPolicy != api.PullAlways {
+			allErrs = append(allErrs, admission.NewForbidden(attributes,
+				field.NotSupported(p.Child("imagePullPolicy"), c.ImagePullPolicy, []string{string(api.PullAlways)}),
+			))
 		}
+	})
+	if len(allErrs) > 0 {
+		// TODO: consider using utilerrors.NewAggregate(allErrs)
```

Comment:
@smarterclayton I didn't want to change the behavior of this plugin without discussing it with someone, but I noticed other plugins use `utilerrors.NewAggregate(allErrs)` rather than returning the first error.

Reply:
Hrm, if we do this then we have to evaluate all images on all pods, vs just the first invalid one.  However, in a successful case we have to check all of them.

I'm ok with the change.

Reply:
But can you open a follow up and we can get the plugin authors to discuss?

Reply:
Opened #79378